storage/aptosdb/src/state_kv_db.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{
+    db_metadata::{DbMetadataKey, DbMetadataSchema, DbMetadataValue},
+    db_options::{gen_state_kv_cfds, state_kv_db_column_families},
+    NUM_STATE_SHARDS,
+};
+use anyhow::Result;
+use aptos_config::config::{RocksdbConfig, RocksdbConfigs};
+use aptos_rocksdb_options::gen_rocksdb_options;
+use aptos_schemadb::{SchemaBatch, DB};
+use aptos_types::transaction::Version;
+use arr_macro::arr;
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+pub const STATE_KV_DB_FOLDER_NAME: &str = "state_kv_db";
+pub const STATE_KV_METADATA_DB_NAME: &str = "state_kv_metadata_db";
+
+pub struct StateKvDb {
+    state_kv_metadata_db: Arc<DB>,
+    _state_kv_db_shards: [Arc<DB>; NUM_STATE_SHARDS],
+}
+
+impl StateKvDb {
+    // TODO(grao): Support more flexible path to make it easier for people to put different shards
+    // on different disks.
+    pub(crate) fn open<P: AsRef<Path>>(
```

Comment:
Let's call it `initialize()` to prevent dumb ass like me from thinking it always opens rocksdb

Reply:
Changed to `new()`.